### ISSUE TITLE
Calling reject() with no argument causes the health check to pass

### ISSUE THREAD

**Issue by bmaupin** [CloudNativeJS/cloud-health#38]
I'm using @cloudnative/health v2.1.1 with Express via @cloudnative/health-connect v2.1.0:

```
$ npm list @cloudnative/health
└─┬ @cloudnative/health-connect@2.1.0
  └── @cloudnative/health@2.1.1
```

I created a readiness promise and noticed it was always returning a 200:

```javascript
  const readinessPromise = () =>
    new Promise(function(resolve, reject) {
      if (global.dbMigrationComplete && global.serverListening) {
        resolve();
      } else {
        reject();
      }
    });
```

Only once I added an argument to `reject()` did the readiness check return a 503:

```javascript
      } else {
        reject(new Error('Database migration not complete or server not started'));
```

Thanks!

**Comment by andrewhughes101** [comment]
Hi @bmaupin I've tried this locally and can reproduce this error, thanks for catching this

**Comment by andrewhughes101** [comment]
@bmaupin Ive created a fix for this in #39 

The error is being produced because the [wrapPromise](https://github.com/CloudNativeJS/cloud-health/blob/d9978bfc3cc677c665133706e65095ed95df403c/src/healthcheck/HealthChecker.ts#L279) method within the Plugin class was trying to assign a statusReason to the value of `error.message` but when you dont throw an Error within the reject statement, `error` in the `wrapPromise` method is undefined. This would lead to the check crashing and 200 being returned instead of 503.

We also recommend that you should pass a reason to your `reject` as shown in the [documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject) 

What do you think of the fix?

**Comment by bmaupin** [comment]
I don't know enough about the code to comment on the fix, but I see someone who does know has added comments to the PR.

I do agree that it's best to pass a reason. But I thought I'd file an issue anyway since it was unexpected behaviour.

Thanks!

**Comment by andrewhughes101** [comment]
@bmaupin we've just released `@cloudnative/health` 2.1.2 on npm. The cloud-health-connect module should pick up this new version

**Comment by bmaupin** [comment]
@andrewhughes101 Perfect, thanks!
